server/src/main/java/org/candlepin/resource/ActivationKeyResource.java

```diff
@@ -279,9 +302,28 @@ public ActivationKey removeProductIdFromKey(
      */
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public List<ActivationKey> findActivationKey() {
-        List<ActivationKey> keyList = activationKeyCurator.listAll();
-        return keyList;
+    public Response findActivationKey() {
+        // TODO: Replace this with use of a cursor/iterator
+        final List<ActivationKey> keyList = activationKeyCurator.listAll();
+        final ObjectMapper mapper = new JsonProvider(true)
+            .locateMapper(Object.class, MediaType.APPLICATION_JSON_TYPE);
+
+        StreamingOutput output = new StreamingOutput() {
+            @Override
+            public void write(OutputStream stream) throws IOException, WebApplicationException {
+                JsonGenerator generator = mapper.getJsonFactory().createGenerator(stream);
+                generator.writeStartArray();
+
+                for (ActivationKey key : keyList) {
+                    mapper.writeValue(generator, new ActivationKeyData(key));
```

Comment:
I haven't tested it, but my initial reaction is that we don't know exactly when mapper converts an object to JSON for output. If it caches stuff until they're flushed or some such, we could end up clobbering a bunch of data.